File: php/features.php
<?php
declare(strict_types=1);

// ParlamentOpenData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ParlamentOpenDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ParlamentOpenDataBaseFeature();
            case "test":
                return new ParlamentOpenDataTestFeature();
            default:
                return new ParlamentOpenDataBaseFeature();
        }
    }
}

File: py/features.py
# ParlamentOpenData SDK feature factory

from feature.base_feature import ParlamentOpenDataBaseFeature
from feature.test_feature import ParlamentOpenDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ParlamentOpenDataBaseFeature(),
        "test": lambda: ParlamentOpenDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
